### PR TITLE
Plugins v2

### PR DESCRIPTION
**Pull request by PhilippLgh**
#### What does it do?
- allows apps to request the start of a service
- fixes the `plugin.execute(command)` functionality
- adds a user config file
- adds a config editor / json editor dialog
- adds a Plugins and Config Menu item
- allows user-defined "registries" i.e. to load additional apps and plugins

#### Any helpful background information?
WIP registry format: https://gist.githubusercontent.com/PhilippLgh/c7745bf029dc1dcb1f41eb732e06b266/raw/cf85e7136be2d1161f39c106d4cba811937ea097/grid-registry.json

example config:
```
{
  "registries": [
    "https://gist.githubusercontent.com/PhilippLgh/c7745bf029dc1dcb1f41eb732e06b266/raw/ccb69ede5f09b20700145681ec255cd250dc9117/grid-registry.json"
  ],
  "plugins": [
    {
      "name": "ipfs",
      "displayName": "IPFS",
      "type": "storage",
      "location": "D:/Projects/Grid/plugins/grid-ipfs-plugin"
    }
  ],
  "apps": [
    {
      "name": "P2P Sharing 2",
      "description": "Example P2P file sharing app with Grid",
      "screenshot": "https://github.com/PhilippLgh/ethereum-grid-tutorials/raw/master/assets/Swarm-file-dropzone.PNG",
      "url": "package://github.com/PhilippLgh/grid-swarm-p2p"
    }
  ]
}
```

#### New dependencies? What are they used for?
- requires new app-manager version that exposes `downloadJson` and includes some fixes
- requires an API change in grid-ui's app overview to show *all* apps

#### Relevant screenshots?
image

### CODE REVIEW

File: Menu.js [hunk 1]
```diff
@@ -25,9 +23,32 @@ module.exports.getMenuTemplate = () => {
...
-        { role: 'minimize' },
-        { role: 'close' }
+        {
+          label: 'rescan',
+          click() {
+            if (!global.PluginHost) {
+              console.log('plugin host not ready')
+              return
+            }
+            global.PluginHost.loadUserRegistries()
```
Comment (PhilippLgh) [inline Menu.js]: this is broken but should be fixed with #296

**Comment by ryanio** [comment]
Updated registry format from philipp: https://gist.githubusercontent.com/PhilippLgh/c7745bf029dc1dcb1f41eb732e06b266/raw/ccb69ede5f09b20700145681ec255cd250dc9117/grid-registry.json

File: Menu.js (hunk 1 above)
Comment (marcgarreau) [reply Menu.js]: @ryanio see ^

**Comment by ryanio** [comment]
Added copy-paste functionality in latest commit if you want to check it works well for you @marcgarreau  

https://github.com/ethereum/grid/pull/293/commits/f1cfacc556f1d3e78f7cb3e613f0bb2b79248eb3, which will close https://github.com/ethereum/grid/issues/299

**Pull request merged**